### ISSUE TITLE
to_featurelayer method fails with "AttributeError: 'float' object has no attribute 'type'" when row has null geometry

### ISSUE THREAD

**Issue by earlmedina** [Esri/arcgis-python-api#1068]
The issue is with the arcgis.features.SpatialDataFrame.to_featurelayer method. As written, it does not properly handle the occurance of null geometry values in a row. 

Example Traceback:
```
AttributeError                            Traceback (most recent call last)
<ipython-input-23-8741f15ecd4a> in <module>
      1 df.spatial.to_featurelayer(
      2     title=fl.properties.name,
----> 3     gis=gis
      4 )

~\AppData\Local\ESRI\conda\envs\pro28\lib\site-packages\arcgis\features\geo\_accessor.py in to_featurelayer(self, title, gis, tags, folder)
   2461                 raise ValueError("GIS object must be provided")
   2462         content = gis.content
-> 2463         return content.import_data(self._data, folder=folder, title=title, tags=tags)
   2464 
   2465     # ----------------------------------------------------------------------

~\AppData\Local\ESRI\conda\envs\pro28\lib\site-packages\arcgis\gis\__init__.py in import_data(self, df, address_fields, folder, item_id, **kwargs)
   6131                 else:
   6132                     ds = df.spatial.to_featureclass(
-> 6133                         location=os.path.join(temp_dir, name)
   6134                     )
   6135                 zip_shp = zipws(path=temp_dir, outfile=temp_zip, keep=False)

~\AppData\Local\ESRI\conda\envs\pro28\lib\site-packages\arcgis\features\geo\_accessor.py in to_featureclass(self, location, overwrite, has_z, has_m, sanitize_columns)
   2399             has_z=has_z,
   2400             sanitize_columns=sanitize_columns,
-> 2401             has_m=has_m,
   2402         )
   2403 

~\AppData\Local\ESRI\conda\envs\pro28\lib\site-packages\arcgis\features\geo\_io\fileops.py in to_featureclass(geo, location, overwrite, validate, sanitize_columns, has_m, has_z)
    850             return res
    851         else:
--> 852             res = _pyshp2(df=df, out_path=out_location, out_name=fc_name)
    853             df.set_index(old_idx)
    854             return res

~\AppData\Local\ESRI\conda\envs\pro28\lib\site-packages\arcgis\features\geo\_io\fileops.py in _pyshp2(df, out_path, out_name)
   1047         for idx, row in df.iterrows():
   1048             geom = row[df.spatial._name]
-> 1049             if hasattr(geom, "type"):
   1050                 if geom.type == "Polygon":
   1051                     shpfile.poly(geom["rings"])

AttributeError: 'float' object has no attribute 'type'
```

In the above example, the issue is found at line 1047 of `arcgis/features/geo/_io/fileops.py`.

As currently written, logic does not allow passing to the else case when a record with null geometry is encountered. In this event, it is possible that the property "type" does not exist altogether, which causes the above AttributeError exception.

It is written like this right now:

```
for idx, row in df.iterrows():
    geom = row[df.spatial._name]
    if geom.type == "Polygon":
        shpfile.poly(geom["rings"])
    elif geom.type == "Polyline":
            shpfile.line(geom["paths"])
    elif geom.type == "Point":
            shpfile.point(x=geom.x, y=geom.y)
    else:
            shpfile.null()
```

But needs to be corrected to this:

```
for idx, row in df.iterrows():  
    geom = row[df.spatial._name]  
    if hasattr(geom, "type"):  
        if geom.type == "Polygon":  
            shpfile.poly(geom["rings"])  
        elif geom.type == "Polyline":  
                shpfile.line(geom["paths"])  
        elif geom.type == "Point":  
                shpfile.point(x=geom.x, y=geom.y)  
        else:  
                shpfile.null()
```

Making the described change resolves the error and the sdf will publish as expected.

**Comment by earlmedina** [comment]
I should note that I just reviewed the output and while it did publish, the result is blank. Maybe the problem is specific to this data, but I do think that particular section of code could be improved with better handling.

**Comment by achapkowski** [comment]
The `SpatialDataFrame` is deprecated and is no longer maintained within the API, please use the Pandas' `accessor` instead.  If you post your code, I can try to help you migrate.

**Comment by earlmedina** [comment]
Sorry, i wrote "arcgis.features.SpatialDataFrame.to_featurelayer" but meant "arcgis.features.GeoAccessor.to_featurelayer."

The issue happens whether you create a spatially enabled dataframe (as described in the introduction documentation) or if you initialize via a feature layer query, using the resulting featureset's sdf property.

**Comment by mohi9282** [comment]
@earlmedina - thank you for reporting the issue. The issue occurs when the backend (geometry engine) is not arcpy. In this case, PyShp is being used to create the shapefile causing issues with null geometries. We will work on fixing this.

**Comment by AtmaMani** [comment]
xref https://github.com/ArcGIS/geosaurus/issues/7005

**Comment by nanaeaubry** [comment]
Fixed in 2.0.0
